### PR TITLE
Update dune-project to generate .opam file.

### PR DESCRIPTION
**Pull request by lyrm**
Correct all issues related to #2.

### CODE REVIEW

**Comment by polytypic** [comment]
LGTM, but I didn't try to build the code.

There is no CI yet?  I assume we'll enable ocaml-ci after transfer to ocaml-multicore.  Also, it would be good to have GitHub actions based CI to run with Windows.

File: backoff.opam [hunk 1]
```diff
@@ -1,15 +1,15 @@
 # This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
-synopsis:
-  "Exponential backoff mechanism for ocaml"
+synopsis: "Exponential backoff mechanism for ocaml"
```
Comment (polytypic) [inline backoff.opam]: Capitalize OCaml?

File: backoff.opam [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 authors: ["KC Sivaramakrishnan <sk826@cl.cam.ac.uk>"]
 license: "ISC"
-homepage: "https://github.com/lyrm/backoff"
-bug-reports: "https://github.com/lyrm/backoff/issues"
+homepage: "https://github.com/ocaml-multicore/backoff"
+bug-reports: "https://github.com/ocaml-multicore/backoff/issues"
 depends: [
   "dune" {>= "3.3"}
-  "ocaml" {>= "5.0"}
+  "ocaml" {>= "4.13"}
```
Comment (polytypic) [inline backoff.opam:11]: Was this the lowest 4.x version that works? I think this could probably build on lower versions.

File: dune-project [hunk 3]
```diff
@@ -1,2 +1,15 @@
...
+(maintainers "Carine Morel <carine@tarides.com>")
+(homepage "https://github.com/ocaml-multicore/backoff")
+(license ISC)
+
+(package
+ (name backoff)
+ (synopsis "Exponential backoff mechanism for ocaml")
+ (depends
+  (ocaml (>= 4.13))
+  (domain_shims (and (>= 0.1.0) :with-test))))
```
Comment (polytypic) [inline dune-project:15]: I think we should also start using `dune fmt` to format these files.  I guess ocaml-ci also has a check for that nowadays.

**Review by polytypic — APPROVED** [review]
I added a couple of nits, but this looks good to me. 🚀

File: backoff.opam (hunk 2 above)
Comment (lyrm) [reply backoff.opam:11]: Mmmh, I did not really think this though about it, I just ... copied/pasted from `kcas` probably ! I will make some checks.

Comment (lyrm) [reply backoff.opam:11]: `domains-shims` works for OCaml from `4.12`, so I guess this may be the lower limit.

Comment (lyrm) [reply backoff.opam:11]: > val min : int -> int -> int
> 
> Return the smaller of the two arguments.
> 
>     Since 4.13.0

Actually `4.13` is right.

Comment (lyrm) [reply backoff.opam:11]: (Also I guess, we could use `Stdlib.min` for older OCaml)

**Pull request merged**